raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -1436,6 +1441,38 @@ public void testInvalidFetchRequest() throws Exception {
         context.assertSentFetchPartitionResponse(Errors.INVALID_REQUEST, epoch, OptionalInt.of(localId));
     }
 
+    @ParameterizedTest
+    @ApiKeyVersionsSource(apiKey = ApiKeys.FETCH)
+    public void testFetchRequestVersionHandling(short version) throws Exception {
```

Comment:
I need to take a deeper look into this one.

Reply:
Updated the UT:
1. With extra comments.
2. Covered the first tryCompleteFetchRequest in the handleFetchRequest